src/app/code/community/IntegerNet/AutoUpsell/Model/Observer.php

```diff
@@ -33,6 +33,19 @@ protected function addItemsFromCategory(RelatedProductCollection $collection, $n
     {
         /** @var Mage_Catalog_Model_Resource_Product_Collection $productsToAdd */
         $productsToAdd = $this->_getProductCategory($product)->getProductCollection();
+        
+        $productsToAdd = Mage::getModel('catalog/product')
+                ->getCollection()
```

Comment:
you are instantiating a new collection here, so we are loosing the category filter